torch/csrc/jit/frontend/schema_type_parser.cpp

```diff
@@ -426,8 +426,8 @@ SchemaTypeParser::parseFakeAndRealType() {
       }
       alias_info = std::move(container);
     } else if (L.nextIf('?')) {
-      fake_value = c10::TypeFactory::create<c10::OptionalType>(fake_value);
-      real_value = c10::TypeFactory::create<c10::OptionalType>(real_value);
+      fake_value = c10::OptionalType::get(fake_value);
+      real_value = c10::OptionalType::get(real_value);
```

Comment:
There is probably a more general fix to this problem, but I just wanted to do something simple that would fix dynamo 

For example, code that calls `TypeFactory::create<ListType>` would probably end up creating multiple "singleton" instances of a given `ListType`, but that doesn't matter in practice because the logic for checking if two list types are the same in `IValue.h` basically just checks if both types are list-like, and then pointer-compares their inner types.